### PR TITLE
Minor arm64 improvements

### PR DESCRIPTION
**Pull request by kd-11**
- Do not add x64 switches to arm64 output. This happens on clang which exports both compilerx86 and compilerarm somehow.
- Do not use x86-specific intrinsics without preprocessor guards. This fixes SPU module compilation allowing SPU llvm to be used in some games.

Most games still hang after loading modules when using llvm. I haven't really checked why and tbh it's not a priority right now.

### CODE REVIEW

**Comment by kd-11** [comment]
You can now play arkedo 01 with SPU llvm. Yay.

**Comment by kd-11** [comment]
I should clarify, this is testing "true" arm64 (i.e linux arm64) not apple's weird arm64 impl. I am running on apple silicon here but basically side-stepping apple's nonsense and just focusing on the hardware.

File: buildfiles/cmake/ConfigureCompiler.cmake [hunk 1]
```diff
@@ -33,16 +33,14 @@ else()
...
-		add_compile_options(-msse -msse2 -mcx16)
-	endif()
-
 	if (COMPILER_ARM)
 		if (APPLE)
 			add_compile_options(-march=armv8.4-a)
 		else()
 			add_compile_options(-march=armv8.1-a)
 		endif()
+	else()
```
Comment (Megamouse) [inline buildfiles/cmake/ConfigureCompiler.cmake:42]: shouldnt this be an elseif (COMPILER_X86) ?

Comment (kd-11) [reply buildfiles/cmake/ConfigureCompiler.cmake:42]: Sure, why not. Maybe someday someone will port to risc-v.

**Pull request merged**